Source/WebCore/page/Page.cpp

```diff
@@ -5305,6 +5305,10 @@ void Page::setSceneIdentifier(String&& sceneIdentifier)
 
 void Page::setObscuredInsets(const FloatBoxExtent& insets)
 {
+    RefPtr localMainFrame = this->localMainFrame();
+    if (RefPtr view = localMainFrame ? localMainFrame->view() : nullptr)
+        view->obscuredInsetsDidChange(insets - m_obscuredInsets);
```

Comment:
Would this be a `WillChange` instead of a `DidChange`?

I also wonder if this needs to be called in the case where `m_obscuredInsets == insets`.

Reply:
I'll change it to `WillChange`